presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -6170,7 +6203,8 @@ public void testApproxMostFrequentWithLong()
 
         MaterializedResult actual2 = computeActual("SELECT approx_most_frequent(2, cast(x as bigint), 15) FROM (values 1, 2, 1, 3, 1, 2, 3, 4, 5) t(x)");
         assertEquals(actual2.getRowCount(), 1);
-        assertEquals(actual2.getMaterializedRows().get(0).getFields().get(0), ImmutableMap.of(1L, 3L, 2L, 2L));
+        Object actual2Value = actual2.getMaterializedRows().get(0).getFields().get(0);
```

Comment:
It seems like Java is always returning the first value for same number of duplicates while C++ picks the last one (and so you changed the results) ?. Can you check the behavior. If yes, then lets create an issue indicating the two sides resolve duplicates differently and document this behavior if need be.

Reply:
Yes that is the reason, although I would need to confirm whether C++ deterministically returns the last value when count is same. I feel it would be better to modify the values in the testcase instead of checking if the result matches either of the two expected rows; something like this:
```
"SELECT approx_most_frequent(2, cast(x as bigint), 15) FROM (values 1, 2, 1, 3, 1, 2, 2, 1, 3, 4, 5) t(x)"
```
Would that be fine?